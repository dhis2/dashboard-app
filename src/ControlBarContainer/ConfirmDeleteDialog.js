import React from 'react';
import PropTypes from 'prop-types';
import { Trans } from 'react-i18next';
import Dialog from 'material-ui/Dialog';
import FlatButton from '../widgets/FlatButton';
import PrimaryButton from '../widgets/PrimaryButton';

const ConfirmDeleteDialog = (
    { dashboardName, onDeleteConfirmed, onContinueEditing, open },
    context
) => {
    const actions = [
        <FlatButton onClick={onDeleteConfirmed}>
            {context.i18n.t('Delete')}
        </FlatButton>,
        <PrimaryButton onClick={onContinueEditing}>
            {context.i18n.t('Continue editing')}
        </PrimaryButton>,
    ];

    return (
        <Dialog
            title={context.i18n.t('Confirm delete dashboard')}
            actions={actions}
            modal={true}
            open={open}
        >
            <Trans>
                Are you sure you want to delete dashboard{' '}
                <strong>{{ dashboardName }}</strong>?
            </Trans>
        </Dialog>
    );
};

ConfirmDeleteDialog.contextTypes = {
    i18n: PropTypes.object,
};

export default ConfirmDeleteDialog;
