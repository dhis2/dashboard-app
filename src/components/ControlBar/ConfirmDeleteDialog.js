import React from 'react';
import i18n from '@dhis2/d2-i18n';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import { Button } from '@dhis2/ui-core';

export const ConfirmDeleteDialog = ({
    dashboardName,
    onDeleteConfirmed,
    onContinueEditing,
    open,
}) => {
    const actions = [
        <Button key="cancel" secondary onClick={onContinueEditing}>
            {i18n.t('Cancel')}
        </Button>,
        <Button key="delete" destructive onClick={onDeleteConfirmed}>
            {i18n.t('Delete')}
        </Button>,
    ];

    return (
        <Dialog open={open} onClose={onContinueEditing}>
            <DialogTitle style={{ fontSize: '1.25rem', fontWeight: '450' }}>
                {i18n.t('Delete dashboard')}
            </DialogTitle>
            <DialogContent>
                <DialogContentText style={{ fontSize: '0.875rem' }}>
                    {i18n.t(
                        'Deleting dashboard "{{ dashboardName }}" will remove it for all users. This action cannot be undone. Are you sure you want to permanently delete this dashboard?',
                        { dashboardName }
                    )}
                </DialogContentText>
            </DialogContent>
            <DialogActions>{actions}</DialogActions>
        </Dialog>
    );
};

export default ConfirmDeleteDialog;
