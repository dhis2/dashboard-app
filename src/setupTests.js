import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const userSettings = {
    keyDbLocale: 'ponyLang',
};

export function getStubContext() {
    return {
        i18n: {
            t: () => {},
        },
        d2: {
            currentUser: {
                firstName: 'Mark the Ghost',
                surname: 'Polak',
                userSettings: {
                    get: key => userSettings[key],
                },
            },
        },
    };
}
