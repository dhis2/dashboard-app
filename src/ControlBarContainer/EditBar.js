import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import ControlBar from 'd2-ui/lib/controlbar/ControlBar';
import Button from 'd2-ui/lib/button/Button';
import TranslationDialog from 'd2-ui/lib/i18n/TranslationDialog.component';
import ConfirmDeleteDialog from './ConfirmDeleteDialog';
import { colors } from '../colors';
import { tSaveDashboard, acClearEditDashboard } from '../actions/editDashboard';
import { tDeleteDashboard } from '../actions/dashboards';
import { sGetEditDashboard } from '../reducers/editDashboard';
import { CONTROL_BAR_ROW_HEIGHT, getOuterHeight } from './ControlBarContainer';
import { apiFetchSelected } from '../api/dashboards';

const styles = {
    save: {
        borderRadius: '2px',
        backgroundColor: colors.royalBlue,
        color: colors.lightGrey,
        fontWeight: '500',
        boxShadow:
            '0 0 2px 0 rgba(0,0,0,0.12), 0 2px 2px 0 rgba(0,0,0,0.24), 0 0 8px 0 rgba(0,0,0,0.12), 0 0 8px 0 rgba(0,0,0,0.24)',
    },
    secondary: {
        color: colors.royalBlue,
        backgroundColor: 'transparent',
        border: 'none',
        fontSize: '14px',
        fontWeight: 500,
        textTransform: 'uppercase',
        padding: '5px',
        height: '36px',
        cursor: 'pointer',
        marginLeft: '10px',
    },
    buttonBar: {
        height: CONTROL_BAR_ROW_HEIGHT,
        paddingTop: '14px',
        marginLeft: '15px',
        marginRight: '15px',
    },
};

class EditBar extends Component {
    state = {
        translationDialogIsOpen: false,
        dashboardModel: undefined,
        confirmDeleteDialogOpen: false,
    };

    onConfirmDelete = () => {
        this.setState({ confirmDeleteDialogOpen: true });
    };

    onContinueEditing = () => {
        this.setState({ confirmDeleteDialogOpen: false });
    };

    onDeleteConfirmed = () => {
        this.setState({ confirmDeleteDialogOpen: false });
        this.props.onDelete(this.props.dashboardId);
    };

    componentDidMount() {
        apiFetchSelected(this.props.dashboardId).then(dashboardModel =>
            this.setState({ dashboardModel })
        );
    }

    toggleTranslationDialog = () => {
        this.setState({
            translationDialogIsOpen: !this.state.translationDialogIsOpen,
        });
    };

    confirmDeleteDialog = () =>
        this.props.deleteAccess && this.props.dashboardId ? (
            <ConfirmDeleteDialog
                dashboardName={this.props.dashboardName}
                onDeleteConfirmed={this.onDeleteConfirmed}
                onContinueEditing={this.onContinueEditing}
                open={this.state.confirmDeleteDialogOpen}
            />
        ) : null;

    translationDialog = () =>
        this.state.dashboardModel && this.state.dashboardModel.id ? (
            <TranslationDialog
                open={this.state.translationDialogIsOpen}
                onRequestClose={this.toggleTranslationDialog}
                objectToTranslate={this.state.dashboardModel}
                fieldsToTranslate={['name', 'description']}
                // TODO handle messages in snackbar
                onTranslationSaved={msg =>
                    console.log('translation update response', msg)
                }
                onTranslationError={err =>
                    console.log('translation update error', err)
                }
            />
        ) : null;

    render() {
        const {
            style,
            onSave,
            onDiscard,
            dashboardId,
            deleteAccess,
        } = this.props;
        const controlBarHeight = getOuterHeight(1, false);

        return (
            <Fragment>
                <ControlBar
                    height={controlBarHeight}
                    editMode={true}
                    expandable={false}
                >
                    <div style={styles.buttonBar}>
                        <div style={style.leftControls}>
                            <Button style={styles.save} onClick={onSave}>
                                Save Changes
                            </Button>
                            {dashboardId && deleteAccess ? (
                                <button
                                    style={styles.secondary}
                                    onClick={this.onConfirmDelete}
                                >
                                    Delete
                                </button>
                            ) : null}
                            {dashboardId ? (
                                <Button
                                    style={styles.secondary}
                                    onClick={this.toggleTranslationDialog}
                                >
                                    Translate
                                </Button>
                            ) : null}
                        </div>
                        <div style={style.rightControls}>
                            <button
                                style={styles.secondary}
                                onClick={onDiscard}
                            >
                                Exit without saving
                            </button>
                        </div>
                    </div>
                </ControlBar>
                {this.translationDialog()}
                {this.confirmDeleteDialog()}
            </Fragment>
        );
    }
}

const mapStateToProps = state => {
    const dashboard = sGetEditDashboard(state);

    return {
        dashboardId: dashboard.id,
        dashboardName: dashboard.name,
        deleteAccess: dashboard.access ? dashboard.access.delete : false,
    };
};

const mapDispatchToProps = dispatch => {
    return {
        onSave: () => {
            dispatch(tSaveDashboard());
        },
        onDiscard: () => {
            dispatch(acClearEditDashboard());
        },
        onDelete: id => {
            dispatch(tDeleteDashboard(id));
        },
    };
};

const EditBarCt = connect(mapStateToProps, mapDispatchToProps)(EditBar);

export default EditBarCt;
