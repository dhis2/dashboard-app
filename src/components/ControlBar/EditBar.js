import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import i18n from '@dhis2/d2-i18n';
import ControlBar from './ControlBar';
import TranslationDialog from '@dhis2/d2-ui-translation-dialog';
import { Button } from '@dhis2/ui-core';

import ConfirmDeleteDialog from './ConfirmDeleteDialog';
import {
    tSaveDashboard,
    acClearEditDashboard,
} from '../../actions/editDashboard';
import {
    tDeleteDashboard,
    acSetDashboardDisplayName,
} from '../../actions/dashboards';
import {
    sGetEditDashboardRoot,
    sGetIsNewDashboard,
} from '../../reducers/editDashboard';
import {
    CONTROL_BAR_ROW_HEIGHT,
    MIN_ROW_COUNT,
    getControlBarHeight,
} from './controlBarDimensions';
import { apiFetchDashboard } from '../../api/dashboards';

import classes from './styles/DashboardsBar.module.css';

const buttonBarStyle = {
    height: CONTROL_BAR_ROW_HEIGHT,
    paddingTop: '15px',
    marginLeft: '15px',
    marginRight: '15px',
};

export class EditBar extends Component {
    state = {
        translationDialogIsOpen: false,
        dashboardModel: undefined,
        confirmDeleteDialogOpen: false,
        redirectUrl: undefined,
    };

    onConfirmDelete = () => {
        this.setState({ confirmDeleteDialogOpen: true });
    };

    onSave = () => {
        this.props.onSave().then(newId => {
            this.setState({ redirectUrl: `/${newId}` });
        });
    };

    onDiscard = () => {
        this.props.onDiscardChanges();
        const redirectUrl = this.props.dashboardId
            ? `/${this.props.dashboardId}`
            : '/';
        this.setState({ redirectUrl });
    };

    onContinueEditing = () => {
        this.setState({ confirmDeleteDialogOpen: false });
    };

    onDeleteConfirmed = () => {
        this.setState({ confirmDeleteDialogOpen: false });
        this.props.onDelete(this.props.dashboardId).then(() => {
            this.setState({ redirectUrl: '/' });
        });
    };

    onTranslationsSaved = async translations => {
        if (translations && translations.length) {
            const dbLocale = await this.context.d2.currentUser.userSettings.get(
                'keyDbLocale'
            );

            const translation = translations.find(
                t => t.locale === dbLocale && t.property === 'NAME'
            );

            if (translation && translation.value) {
                this.props.onTranslate(
                    this.props.dashboardId,
                    translation.value
                );
            }
        }
    };

    fetchDashboardModel = () => {
        if (this.props.dashboardId && !this.state.dashboardModel) {
            apiFetchDashboard(this.props.dashboardId).then(dashboardModel =>
                this.setState({ dashboardModel })
            );
        }
    };

    componentDidMount() {
        this.fetchDashboardModel();
    }

    componentDidUpdate() {
        this.fetchDashboardModel();
    }

    toggleTranslationDialog = () => {
        this.setState({
            translationDialogIsOpen: !this.state.translationDialogIsOpen,
        });
    };

    confirmDeleteDialog = () =>
        this.props.deleteAccess && this.props.dashboardId ? (
            <ConfirmDeleteDialog
                dashboardName={this.props.dashboardName}
                onDeleteConfirmed={this.onDeleteConfirmed}
                onContinueEditing={this.onContinueEditing}
                open={this.state.confirmDeleteDialogOpen}
            />
        ) : null;

    translationDialog = () =>
        this.state.dashboardModel && this.state.dashboardModel.id ? (
            <TranslationDialog
                className="translation-dialog"
                d2={this.context.d2}
                open={this.state.translationDialogIsOpen}
                onRequestClose={this.toggleTranslationDialog}
                objectToTranslate={this.state.dashboardModel}
                fieldsToTranslate={['name', 'description']}
                // TODO handle messages in snackbar
                onTranslationSaved={this.onTranslationsSaved}
                onTranslationError={err =>
                    console.log('translation update error', err)
                }
            />
        ) : null;

    render() {
        if (this.state.redirectUrl) {
            return <Redirect to={this.state.redirectUrl} />;
        }

        const { dashboardId, deleteAccess, updateAccess } = this.props;
        const controlBarHeight = getControlBarHeight(MIN_ROW_COUNT);

        const discardBtnText = updateAccess
            ? i18n.t('Exit without saving')
            : i18n.t('Go to dashboards');

        return (
            <Fragment>
                <ControlBar height={controlBarHeight} editMode={true}>
                    <div style={buttonBarStyle}>
                        {updateAccess ? (
                            <div className={classes.leftControls}>
                                <span style={{ marginRight: '15px' }}>
                                    <Button primary onClick={this.onSave}>
                                        {i18n.t('Save changes')}
                                    </Button>
                                </span>

                                {dashboardId ? (
                                    <span style={{ marginRight: '15px' }}>
                                        <Button
                                            onClick={
                                                this.toggleTranslationDialog
                                            }
                                        >
                                            {i18n.t('Translate')}
                                        </Button>
                                    </span>
                                ) : null}
                                {dashboardId && deleteAccess ? (
                                    <Button onClick={this.onConfirmDelete}>
                                        {i18n.t('Delete')}
                                    </Button>
                                ) : null}
                            </div>
                        ) : null}

                        <div className={classes.rightControls}>
                            <Button secondary onClick={this.onDiscard}>
                                {discardBtnText}
                            </Button>
                        </div>
                    </div>
                </ControlBar>
                {this.translationDialog()}
                {this.confirmDeleteDialog()}
            </Fragment>
        );
    }
}

EditBar.contextTypes = {
    d2: PropTypes.object,
};

const mapStateToProps = state => {
    const dashboard = sGetEditDashboardRoot(state);

    let deleteAccess;
    let updateAccess;
    if (sGetIsNewDashboard(state)) {
        deleteAccess = true;
        updateAccess = true;
    } else {
        updateAccess = dashboard.access ? dashboard.access.update : false;
        deleteAccess = dashboard.access ? dashboard.access.delete : false;
    }

    return {
        dashboardId: dashboard.id,
        dashboardName: dashboard.name,
        deleteAccess,
        updateAccess,
    };
};

const mapDispatchToProps = dispatch => ({
    onSave: () => dispatch(tSaveDashboard()).then(id => id),
    onDelete: id => dispatch(tDeleteDashboard(id)),
    onDiscardChanges: () => dispatch(acClearEditDashboard()),
    onTranslate: (id, value) => dispatch(acSetDashboardDisplayName(id, value)),
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(EditBar);
