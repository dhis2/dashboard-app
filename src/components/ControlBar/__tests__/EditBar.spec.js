import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Button } from '@dhis2/ui'

import { EditBar } from '../EditBar'
import ConfirmDeleteDialog from '../ConfirmDeleteDialog'
import { getStubContext } from '../../../setupTests'

const mockDashboardModels = {
    rainbow: {
        id: 'rainbow123',
    },
}

jest.mock('../../../api/dashboards', () => ({
    apiFetchDashboard: (dataEngine, id) =>
        Promise.resolve(mockDashboardModels[id]),
}))

jest.mock('@dhis2/d2-ui-translation-dialog', () => () => {
    return <div className="mock-dhis2-translation-dialog" />
})

describe('EditBar', () => {
    let props
    let shallowEditBar
    const editBar = () => {
        const context = getStubContext()

        if (!shallowEditBar) {
            shallowEditBar = shallow(<EditBar {...props} />, {
                context,
            })
        }
        return shallowEditBar
    }

    beforeEach(() => {
        props = {
            style: {},
            onSave: undefined,
            onDiscardChanges: undefined,
            dashboardId: undefined,
            deleteAccess: undefined,
            updateAccess: undefined,
        }
        shallowEditBar = undefined
    })

    it('renders the EditBar', () => {
        expect(toJson(editBar())).toMatchSnapshot()
    })

    describe('when update access is false', () => {
        beforeEach(() => {
            props.updateAccess = false
        })

        it('renders only the Discard button', () => {
            expect(toJson(editBar())).toMatchSnapshot()
        })
    })

    describe('when update access is true', () => {
        beforeEach(() => {
            props.updateAccess = true
        })

        describe('when no dashboard id property', () => {
            beforeEach(() => {
                props.onSave = jest.fn(() => ({ then: () => {} }))
                props.onDiscardChanges = jest.fn()
            })

            it('renders Save and Discard buttons but no dialogs', () => {
                const wrapper = editBar()

                return Promise.resolve().then(() => {
                    expect(toJson(wrapper)).toMatchSnapshot()
                })
            })

            it('triggers the save action', () => {
                editBar()
                    .find(Button)
                    .filterWhere(n => n.childAt(0).text() === 'Save changes')
                    .simulate('click')

                expect(props.onSave).toHaveBeenCalled()
            })

            it('triggers the discard action', () => {
                editBar()
                    .find(Button)
                    .filterWhere(
                        n => n.childAt(0).text() === 'Exit without saving'
                    )
                    .simulate('click')
                expect(props.onDiscardChanges).toHaveBeenCalled()
            })
        })

        describe('when dashboard id property provided', () => {
            beforeEach(() => {
                props.dashboardId = 'rainbow'
            })

            it('renders Save, Translate and Discard buttons and the TranslationDialog but not ConfirmDeleteDialog', () => {
                const wrapper = editBar()

                return Promise.resolve().then(() => {
                    expect(toJson(wrapper)).toMatchSnapshot()
                })
            })

            describe('when TRANSLATE button is clicked', () => {
                const getAsyncWrapper = () => {
                    const wrapper = editBar()

                    return Promise.resolve().then(() => {
                        wrapper.update()
                        return wrapper
                    })
                }

                it('shows the translate dialog', done => {
                    getAsyncWrapper().then(wrapper => {
                        expect(toJson(wrapper)).toMatchSnapshot()

                        wrapper
                            .find(Button)
                            .filterWhere(
                                n => n.childAt(0).text() === 'Translate'
                            )
                            .simulate('click')

                        expect(toJson(wrapper)).toMatchSnapshot()
                        done()
                    })
                })

                describe('when translations saved', () => {
                    beforeEach(() => {
                        props.onTranslate = jest.fn()
                    })

                    it('triggers onTranslationsSaved', done => {
                        getAsyncWrapper()
                            .then(wrapper => {
                                wrapper
                                    .find('.translation-dialog')
                                    .simulate('translationSaved', [
                                        {
                                            locale: 'ponyLang',
                                            property: 'NAME',
                                            value: 'Regnbue',
                                        },
                                    ])
                            })
                            .then(() => {
                                expect(props.onTranslate).toHaveBeenCalled()
                                expect(props.onTranslate).toHaveBeenCalledWith(
                                    'rainbow',
                                    'Regnbue'
                                )
                                done()
                            })
                    })
                })
            })

            describe('when deleteAccess is true', () => {
                beforeEach(() => {
                    props.deleteAccess = true
                    props.onDelete = jest.fn(() => ({ then: () => {} }))
                })

                it('renders Translate, Delete, and Discard buttons and ConfirmDeleteDialog', () => {
                    expect(toJson(editBar())).toMatchSnapshot()
                })

                it('shows the confirm delete dialog when delete button clicked', () => {
                    const wrapper = editBar()
                    expect(
                        wrapper.find(ConfirmDeleteDialog).prop('open')
                    ).toEqual(false)

                    wrapper
                        .find(Button)
                        .filterWhere(n => n.childAt(0).text() === 'Delete')
                        .simulate('click')

                    expect(
                        wrapper.find(ConfirmDeleteDialog).prop('open')
                    ).toEqual(true)
                    expect(props.onDelete).not.toHaveBeenCalled()
                })

                it('triggers onDelete when delete confirmed', () => {
                    const dlg = editBar().find(ConfirmDeleteDialog)
                    dlg.simulate('deleteConfirmed')

                    expect(dlg.prop('open')).toEqual(false)
                    expect(props.onDelete).toHaveBeenCalled()
                })

                it('does not trigger onDelete when delete not confirmed', () => {
                    const dlg = editBar().find(ConfirmDeleteDialog)
                    dlg.simulate('continueEditing')

                    expect(dlg.prop('open')).toEqual(false)
                    expect(props.onDelete).not.toHaveBeenCalled()
                })
            })
        })
    })
})
