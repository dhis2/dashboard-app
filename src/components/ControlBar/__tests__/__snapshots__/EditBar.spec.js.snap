// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditBar renders the EditBar 1`] = `
<Fragment>
  <ControlBar
    editMode={true}
    height={64}
    onChangeHeight={null}
    onEndDrag={null}
  >
    <div
      style={
        Object {
          "height": 40,
          "marginLeft": "15px",
          "marginRight": "15px",
          "paddingTop": "15px",
        }
      }
    >
      <div
        className="rightControls"
      >
        <Button
          dataTest="dhis2-uicore-button"
          onClick={[Function]}
          secondary={true}
          type="button"
        >
          Go to dashboards
        </Button>
      </div>
    </div>
  </ControlBar>
</Fragment>
`;

exports[`EditBar when update access is false renders only the Discard button 1`] = `
<Fragment>
  <ControlBar
    editMode={true}
    height={64}
    onChangeHeight={null}
    onEndDrag={null}
  >
    <div
      style={
        Object {
          "height": 40,
          "marginLeft": "15px",
          "marginRight": "15px",
          "paddingTop": "15px",
        }
      }
    >
      <div
        className="rightControls"
      >
        <Button
          dataTest="dhis2-uicore-button"
          onClick={[Function]}
          secondary={true}
          type="button"
        >
          Go to dashboards
        </Button>
      </div>
    </div>
  </ControlBar>
</Fragment>
`;

exports[`EditBar when update access is true when dashboard id property provided renders Save, Translate and Discard buttons and the TranslationDialog but not ConfirmDeleteDialog 1`] = `
<Fragment>
  <ControlBar
    editMode={true}
    height={64}
    onChangeHeight={null}
    onEndDrag={null}
  >
    <div
      style={
        Object {
          "height": 40,
          "marginLeft": "15px",
          "marginRight": "15px",
          "paddingTop": "15px",
        }
      }
    >
      <div
        className="leftControls"
      >
        <span
          style={
            Object {
              "marginRight": "15px",
            }
          }
        >
          <Button
            dataTest="dhis2-uicore-button"
            onClick={[Function]}
            primary={true}
            type="button"
          >
            Save changes
          </Button>
        </span>
        <span
          style={
            Object {
              "marginRight": "15px",
            }
          }
        >
          <Button
            dataTest="dhis2-uicore-button"
            onClick={[Function]}
            type="button"
          >
            Translate
          </Button>
        </span>
      </div>
      <div
        className="rightControls"
      >
        <Button
          dataTest="dhis2-uicore-button"
          onClick={[Function]}
          secondary={true}
          type="button"
        >
          Exit without saving
        </Button>
      </div>
    </div>
  </ControlBar>
  <Component
    className="translation-dialog"
    d2={
      Object {
        "currentUser": Object {
          "firstName": "Mark the Ghost",
          "surname": "Polak",
          "userSettings": Object {
            "get": [Function],
          },
        },
      }
    }
    fieldsToTranslate={
      Array [
        "name",
        "description",
      ]
    }
    objectToTranslate={
      Object {
        "id": "rainbow123",
      }
    }
    onRequestClose={[Function]}
    onTranslationError={[Function]}
    onTranslationSaved={[Function]}
    open={false}
  />
</Fragment>
`;

exports[`EditBar when update access is true when dashboard id property provided when TRANSLATE button is clicked shows the translate dialog 1`] = `
<Fragment>
  <ControlBar
    editMode={true}
    height={64}
    onChangeHeight={null}
    onEndDrag={null}
  >
    <div
      style={
        Object {
          "height": 40,
          "marginLeft": "15px",
          "marginRight": "15px",
          "paddingTop": "15px",
        }
      }
    >
      <div
        className="leftControls"
      >
        <span
          style={
            Object {
              "marginRight": "15px",
            }
          }
        >
          <Button
            dataTest="dhis2-uicore-button"
            onClick={[Function]}
            primary={true}
            type="button"
          >
            Save changes
          </Button>
        </span>
        <span
          style={
            Object {
              "marginRight": "15px",
            }
          }
        >
          <Button
            dataTest="dhis2-uicore-button"
            onClick={[Function]}
            type="button"
          >
            Translate
          </Button>
        </span>
      </div>
      <div
        className="rightControls"
      >
        <Button
          dataTest="dhis2-uicore-button"
          onClick={[Function]}
          secondary={true}
          type="button"
        >
          Exit without saving
        </Button>
      </div>
    </div>
  </ControlBar>
  <Component
    className="translation-dialog"
    d2={
      Object {
        "currentUser": Object {
          "firstName": "Mark the Ghost",
          "surname": "Polak",
          "userSettings": Object {
            "get": [Function],
          },
        },
      }
    }
    fieldsToTranslate={
      Array [
        "name",
        "description",
      ]
    }
    objectToTranslate={
      Object {
        "id": "rainbow123",
      }
    }
    onRequestClose={[Function]}
    onTranslationError={[Function]}
    onTranslationSaved={[Function]}
    open={false}
  />
</Fragment>
`;

exports[`EditBar when update access is true when dashboard id property provided when TRANSLATE button is clicked shows the translate dialog 2`] = `
<Fragment>
  <ControlBar
    editMode={true}
    height={64}
    onChangeHeight={null}
    onEndDrag={null}
  >
    <div
      style={
        Object {
          "height": 40,
          "marginLeft": "15px",
          "marginRight": "15px",
          "paddingTop": "15px",
        }
      }
    >
      <div
        className="leftControls"
      >
        <span
          style={
            Object {
              "marginRight": "15px",
            }
          }
        >
          <Button
            dataTest="dhis2-uicore-button"
            onClick={[Function]}
            primary={true}
            type="button"
          >
            Save changes
          </Button>
        </span>
        <span
          style={
            Object {
              "marginRight": "15px",
            }
          }
        >
          <Button
            dataTest="dhis2-uicore-button"
            onClick={[Function]}
            type="button"
          >
            Translate
          </Button>
        </span>
      </div>
      <div
        className="rightControls"
      >
        <Button
          dataTest="dhis2-uicore-button"
          onClick={[Function]}
          secondary={true}
          type="button"
        >
          Exit without saving
        </Button>
      </div>
    </div>
  </ControlBar>
  <Component
    className="translation-dialog"
    d2={
      Object {
        "currentUser": Object {
          "firstName": "Mark the Ghost",
          "surname": "Polak",
          "userSettings": Object {
            "get": [Function],
          },
        },
      }
    }
    fieldsToTranslate={
      Array [
        "name",
        "description",
      ]
    }
    objectToTranslate={
      Object {
        "id": "rainbow123",
      }
    }
    onRequestClose={[Function]}
    onTranslationError={[Function]}
    onTranslationSaved={[Function]}
    open={true}
  />
</Fragment>
`;

exports[`EditBar when update access is true when dashboard id property provided when deleteAccess is true renders Translate, Delete, and Discard buttons and ConfirmDeleteDialog 1`] = `
<Fragment>
  <ControlBar
    editMode={true}
    height={64}
    onChangeHeight={null}
    onEndDrag={null}
  >
    <div
      style={
        Object {
          "height": 40,
          "marginLeft": "15px",
          "marginRight": "15px",
          "paddingTop": "15px",
        }
      }
    >
      <div
        className="leftControls"
      >
        <span
          style={
            Object {
              "marginRight": "15px",
            }
          }
        >
          <Button
            dataTest="dhis2-uicore-button"
            onClick={[Function]}
            primary={true}
            type="button"
          >
            Save changes
          </Button>
        </span>
        <span
          style={
            Object {
              "marginRight": "15px",
            }
          }
        >
          <Button
            dataTest="dhis2-uicore-button"
            onClick={[Function]}
            type="button"
          >
            Translate
          </Button>
        </span>
        <Button
          dataTest="dhis2-uicore-button"
          onClick={[Function]}
          type="button"
        >
          Delete
        </Button>
      </div>
      <div
        className="rightControls"
      >
        <Button
          dataTest="dhis2-uicore-button"
          onClick={[Function]}
          secondary={true}
          type="button"
        >
          Exit without saving
        </Button>
      </div>
    </div>
  </ControlBar>
  <ConfirmDeleteDialog
    onContinueEditing={[Function]}
    onDeleteConfirmed={[Function]}
    open={false}
  />
</Fragment>
`;

exports[`EditBar when update access is true when no dashboard id property renders Save and Discard buttons but no dialogs 1`] = `
<Fragment>
  <ControlBar
    editMode={true}
    height={64}
    onChangeHeight={null}
    onEndDrag={null}
  >
    <div
      style={
        Object {
          "height": 40,
          "marginLeft": "15px",
          "marginRight": "15px",
          "paddingTop": "15px",
        }
      }
    >
      <div
        className="leftControls"
      >
        <span
          style={
            Object {
              "marginRight": "15px",
            }
          }
        >
          <Button
            dataTest="dhis2-uicore-button"
            onClick={[Function]}
            primary={true}
            type="button"
          >
            Save changes
          </Button>
        </span>
      </div>
      <div
        className="rightControls"
      >
        <Button
          dataTest="dhis2-uicore-button"
          onClick={[Function]}
          secondary={true}
          type="button"
        >
          Exit without saving
        </Button>
      </div>
    </div>
  </ControlBar>
</Fragment>
`;
